2a_model/src/models/Snakefile_base.smk

```diff
@@ -27,6 +27,46 @@ rule as_run_config:
         asRunConfig(config,output[0])
 
 
+rule prep_io_data:
+    input:
+        "../../../out/well_obs_inputs.zarr",
+        "../../../out/well_obs_targets.zarr",
+    output:
+        "{outdir}/prepped.npz"
+    run:
+        prep_all_data(x_data_file=input[0],
+                      y_data_file=input[1],
+                      x_vars=config['x_vars'],
+                      y_vars_finetune=config['y_vars'],
+                      spatial_idx_name='site_id',
+                      time_idx_name='date',
+                      train_start_date=config['train_start_date'],
+                      train_end_date=config['train_end_date'],
+                      val_start_date=config['val_start_date'],
+                      val_end_date=config['val_end_date'],
+                      test_start_date=config['test_start_date'],
+                      test_end_date=config['test_end_date'],
+                      val_sites=config['validation_sites'],
+                      out_file=output[0],
+                      normalize_y=False,
+                      trn_offset = config['trn_offset'],
+                      tst_val_offset = config['tst_val_offset'])
+
+        # check to make sure there is no validation or testing data
+        # in the training data set
+        data = np.load(output[0], allow_pickle=True)
+        df_trn = prepped_array_to_df(data['y_obs_trn'],
+                                     data['times_trn'],
+                                     data['ids_trn'],
+                                     col_names=data['y_obs_vars'])
+        df_trn_val_sites = df_trn[df_trn.seg_id_nat.isin(config['validation_sites'])]
+        df_trn_tst_sites = df_trn[df_trn.seg_id_nat.isin(config['test_sites'])]
+
+        assert df_trn_val_sites['do_mean'].notna().sum() == 0
+        assert df_trn_tst_sites['do_mean'].notna().sum() == 0
```

Comment:
These lines are here now to make sure that there is no data in the training dataset for the validation sites. If there is data in any of the sites, it will throw an error. I've tested this and it works.